lib/utils.js

```diff
@@ -53,3 +53,31 @@ export async function loadGetInitialProps (Component, ctx) {
   }
   return props
 }
+
+export class LockManager {
```

Comment:
I think we can use `https://github.com/timdp/es6-promise-pool` or other existing modules. I feel the idea of lock doesn't fit to JS :D